sources/Valkey.Glide/BaseClient.HashCommands.cs

```diff
@@ -55,12 +55,72 @@ public async Task<bool> HashExistsAsync(ValkeyKey key, ValkeyValue hashField, Co
         return await Command(Request.HashExistsAsync(key, hashField));
     }
 
+    public async Task<long> HashIncrementAsync(ValkeyKey key, ValkeyValue hashField, long value = 1, CommandFlags flags = CommandFlags.None)
+    {
+        Utils.Requires<NotImplementedException>(flags == CommandFlags.None, "Command flags are not supported by GLIDE");
+        return await Command(Request.HashIncrementAsync(key, hashField, value));
+    }
+
+    public async Task<double> HashIncrementAsync(ValkeyKey key, ValkeyValue hashField, double value, CommandFlags flags = CommandFlags.None)
+    {
+        Utils.Requires<NotImplementedException>(flags == CommandFlags.None, "Command flags are not supported by GLIDE");
+        return await Command(Request.HashIncrementAsync(key, hashField, value));
+    }
+
+    public async Task<ValkeyValue[]> HashKeysAsync(ValkeyKey key, CommandFlags flags = CommandFlags.None)
+    {
+        Utils.Requires<NotImplementedException>(flags == CommandFlags.None, "Command flags are not supported by GLIDE");
+        return await Command(Request.HashKeysAsync(key));
+    }
+
     public async Task<long> HashLengthAsync(ValkeyKey key, CommandFlags flags = CommandFlags.None)
     {
         Utils.Requires<NotImplementedException>(flags == CommandFlags.None, "Command flags are not supported by GLIDE");
         return await Command(Request.HashLengthAsync(key));
     }
 
+    public async IAsyncEnumerable<HashEntry> HashScanAsync(ValkeyKey key, ValkeyValue pattern = default, int pageSize = 250, long cursor = 0, int pageOffset = 0, CommandFlags flags = CommandFlags.None)
+    {
+        Utils.Requires<NotImplementedException>(flags == CommandFlags.None, "Command flags are not supported by GLIDE");
+
+        long currentCursor = cursor;
+
+        do
+        {
+            (long nextCursor, HashEntry[] entries) = await Command(Request.HashScanAsync(key, currentCursor, pattern, pageSize));
+
+            IEnumerable<HashEntry> entriesToYield = pageOffset > 0 ? entries.Skip(pageOffset) : entries;
+
+            foreach (HashEntry entry in entriesToYield)
+            {
+                yield return entry;
+            }
+
+            currentCursor = nextCursor;
+        } while (currentCursor != 0);
+    }
+
+    public async IAsyncEnumerable<ValkeyValue> HashScanNoValuesAsync(ValkeyKey key, ValkeyValue pattern = default, int pageSize = 250, long cursor = 0, int pageOffset = 0, CommandFlags flags = CommandFlags.None)
+    {
+        Utils.Requires<NotImplementedException>(flags == CommandFlags.None, "Command flags are not supported by GLIDE");
+
+        long currentCursor = cursor;
+
+        do
+        {
+            (long nextCursor, ValkeyValue[] fields) = await Command(Request.HashScanNoValuesAsync(key, currentCursor, pattern, pageSize));
```

Comment:
```suggestion
            (long nextCursor, ValkeyValue[] fields) = await Command(Request.HashScan(key, currentCursor, pattern, pageSize, true));
```
I think it should be the same request, just one more param set

Reply:
BTW, we don't need `Async` in request names, but we can do bulk rename later